src/dvclive/dvc.py

```diff
@@ -177,3 +181,29 @@ def get_dvc_stage_template(live):
     dvcyaml = output.getvalue()
     output.close()
     return dvcyaml
+
+
+def update_models(path, attrs, models):
+    from dvc.repo.artifacts import name_is_compatible
+
+    if attrs["name"] is None:
+        name = find_unused_name(path, models)
+    else:
+        name = attrs.pop("name")
+
+    attrs["path"] = path
+    if name_is_compatible(name):
+        if name not in models:
+            models[name] = {}
+        for k, v in attrs.items():
+            if v is not None:
+                models[name][k] = v
```

Comment:
I think we should overwrite and leave this up to the user to manage, so we could reduce this to `models[name] = attrs`.

Reply:
The thing is that `attrs` here is coming from the explicit optional arguments in `log_artifact` that we have for each of the attributes. 

So, here or elsewhere we need a loop like this to discard `None` values, unless we are ok with dumping to `artifacts.yaml` having attributes set to None.

(Not having to handle this was one of the "pros" of `kwargs`)